packages/glamor/src/StyleSheet.ts

```diff
@@ -0,0 +1,213 @@
+                    const tag = last(this.tags);
+                    tag.insertBefore(document.createTextNode(rule), tag.firstChild);
+                } else {
+                    last(this.tags).appendChild(document.createTextNode(rule));
+                }
+            }
+        } else {
+            // server side is pretty simple
+            this.sheet.insertRule(rule, rule.indexOf('@import') !== -1);
+        }
+
+        this.ruleCounter++;
+        if (isBrowser && this.ruleCounter % this.maxRules === 0) {
+            this.tags.push(makeStyleTag());
+        }
+
+        return this.ruleCounter - 1;
+    }
+
+    // delete(index) {
+    //  // we insert a blank rule when 'deleting' so previously returned indexes remain stable
+    //   return this.replace(index, '');
+    // }
+
+    flush() {
+        this.injected = false;
+
+        if (isBrowser) {
+            this.tags.forEach(tag => tag.parentNode.removeChild(tag));
+            this.tags = [];
+            this.sheet = null;
+            this.ruleCounter = 0;
+            // todo - look for remnants in document.styleSheets
+        } else {
+            // simpler on server
+            //this.sheet.emptyCssRules();
+            this.sheet.emptyCssRules();
+        }
+    }
+
+    rules() {
+        if (!isBrowser) {
+            return this.sheet.getCSSRules();
+        }
+
+        let arr: Array<CSSRule> = [];
+
+        this.tags.forEach(tag => arr.splice(arr.length, 0, ...Array.from(sheetForTag(tag).cssRules)));
+
+        return arr;
+    }
+
+    /**
+     * Insert a new css rule into the <style> tag when it's in the browser environment 
+     * @param rule 
+     */
+    private browInsert(rule: string) {
+        // this weirdness for perf, and chrome's weird bug
+        // https://stackoverflow.com/questions/20007992/chrome-suddenly-stopped-accepting-insertrule
+        try {
+            let sheet = this.getSheet();
+            sheet.insertRule(rule, rule.indexOf('@import') !== -1 ? 0 : sheet.cssRules.length);
+        } catch (e) {
+            if (isDev) {
+                // might need beter dx for this
+                console.warn('whoops, illegal rule inserted', rule); // eslint-disable-line no-console
+            }
+        }
+    }
+}
+
+function makeStyleTag() {
+    let tag = document.createElement('style');
+    tag.type = 'text/css';
+    tag.setAttribute('data-glamor', '');
+    tag.appendChild(document.createTextNode(''));
+    (document.head || document.getElementsByTagName('head')[0]).appendChild(tag);
+    return tag;
+}
+
+function last<T>(arr: Array<T>): T {
+    return arr[arr.length - 1];
+}
+
+function sheetForTag(tag: HTMLStyleElement) {
+    if (tag.sheet) {
+        return tag.sheet as CSSStyleSheet;
+    }
+
+    // this weirdness brought to you by firefox
+    // tslint:disable-next-line:prefer-for-of
+    for (let i = 0; i < document.styleSheets.length; i++) {
+        if (document.styleSheets[i].ownerNode === tag) {
```

Comment:
minor nit, seems like the PR is inconsistent on spacing; 2 vs 4... (personally prefer 2 :)